lib/libzfs/libzfs_dataset.c

```diff
@@ -1459,7 +1459,6 @@ zfs_valid_proplist(libzfs_handle_t *hdl, zfs_type_t type, nvlist_t *nvl,
 
 			switch (prop) {
 			case ZFS_PROP_RESERVATION:
```

Comment:
This whole case should go, reservations make even less sense than refreservations here.